limit = 15

# method definition is self contained - does not have access to outside local variables unless those variables are passed in
# def fib(first_num, second_num)
#   while first_num + second_num < limit
#     sum = first_num + second_num
#     first_num = second_num
#     second_num = sum
#   end
#   sum
# end

# result = fib(0, 1) # the limit variable in the outer scope is not passed in
# # calling fuction will throw an undefined local variable error
# puts "result is #{result}"

# code fix
def fib(first_num, second_num, limit)
  while first_num + second_num < limit
    sum = first_num + second_num
    first_num = second_num
    second_num = sum
  end
  sum
end

result = fib(0, 1, limit)
puts "result is #{result}"
